Sources/MapboxNavigation/NavigationCameraType.swift

```diff
@@ -0,0 +1,16 @@
+/**
+ Possible types of `NavigationCamera`.
+ */
+public enum NavigationCameraType {
```

Comment:
Since camera behaviors are different on iOS and CarPlay there is a need to distinguish `NavigationMapView`s which are created for each platform (because they hold `NavigationCamera` instance). Feel free to propose better naming if you have any better ideas.

Reply:
I was thinking about this. If the only use for now is CarPlay then my sense is to call that out specifically and call this .carPlay (or similar). I don't think our developers would necessarily know what a "head unit" is it may be confusing.
If we are already using the notion of a "head unit" in other parts of the SDK then maybe it will be clear, but as far as I know, the opposite is true: We reference CarPlay in many parts of the SDK and don't refer to a "head unit" anywhere else.

Reply:
Yep, I thought that `headUnit` would be more generic and not really dependent on platform branding. At the same time `CarPlay` sounds more exact and clear. I think I'll use `.carPlay` and `.mobile` for `NavigationCameraType`.

Reply:
`iPhone` would be a fine alternative to `mobile`, since an iPad-optimized camera could conceivably be closer to a CarPlay-optimized camera than an iPhone-optimized one.

Reply:
I was thinking about `.iOS`, but since `CarPlay` is just a part of `iOS` it'd contradict each other. `.iPhone` sounds a bit too restrictive IMO, at this point there is technically no difference between cameras on `iPhone` and `iPad`.

Reply:
The difference is that an iPad typically has a different form factor, like the way that CarPlay typically has a different form factor. I could picture the camera behaving differently for different form factors, but then again, maybe size classes would be a better way to accomplish that.